dashboard/modules/reporter/reporter_head.py

```diff
@@ -24,58 +23,32 @@ class ReportHead(dashboard_utils.DashboardHeadModule):
     def __init__(self, dashboard_head):
         super().__init__(dashboard_head)
         self._stubs = {}
-        self._profiling_stats = {}
         DataSource.agents.signal.append(self._update_stubs)
 
     async def _update_stubs(self, change):
+        if change.old:
+            ip, port = next(iter(change.old.items()))
+            self._stubs.pop(ip)
         if change.new:
             ip, ports = next(iter(change.new.items()))
             channel = aiogrpc.insecure_channel("{}:{}".format(ip, ports[1]))
             stub = reporter_pb2_grpc.ReporterServiceStub(channel)
             self._stubs[ip] = stub
-        if change.old:
-            ip, port = next(iter(change.old.items()))
-            self._stubs.pop(ip)
 
     @routes.get("/api/launch_profiling")
     async def launch_profiling(self, req) -> aiohttp.web.Response:
-        node_id = req.query.get("node_id")
+        ip = req.query.get("ip")
         pid = int(req.query.get("pid"))
         duration = int(req.query.get("duration"))
-        profiling_id = str(uuid.uuid4())
-        reporter_stub = self._stubs[node_id]
+        reporter_stub = self._stubs[ip]
         reply = await reporter_stub.GetProfilingStats(
             reporter_pb2.GetProfilingStatsRequest(pid=pid, duration=duration))
-        self._profiling_stats[profiling_id] = reply
-        return await dashboard_utils.rest_response(
-            success=True,
-            message="Profiling launched.",
-            profiling_id=profiling_id)
-
-    @routes.get("/api/check_profiling_status")
-    async def check_profiling_status(self, req) -> aiohttp.web.Response:
-        profiling_id = req.query.get("profiling_id")
-        is_present = profiling_id in self._profiling_stats
-        if not is_present:
-            status = {"status": "pending"}
-        else:
-            reply = self._profiling_stats[profiling_id]
-            if reply.stderr:
-                status = {"status": "error", "error": reply.stderr}
-            else:
-                status = {"status": "finished"}
-        return await dashboard_utils.rest_response(
-            success=True, message="Profiling status fetched.", status=status)
-
-    @routes.get("/api/get_profiling_info")
-    async def get_profiling_info(self, req) -> aiohttp.web.Response:
-        profiling_id = req.query.get("profiling_id")
-        profiling_stats = self._profiling_stats.get(profiling_id)
-        assert profiling_stats, "profiling not finished"
+        print("ccc", reply.std_err, reply.std_out)
         return await dashboard_utils.rest_response(
             success=True,
-            message="Profiling info fetched.",
-            profiling_info=json.loads(profiling_stats.profiling_stats))
+            message="Profiling success.",
+            profiling_info=json.loads(reply.profiling_stats)
+            if reply.profiling_stats else reply.std_out)
```

Comment:
```suggestion
                if reply.profiling_stats else reply.std_out)
```

Reply:
I will assign the conditional expression to a variable.